Cache pokes in core

peterzen opened the pull request:
Depends on #2624
Supersedes #2660

martonp reviewed (commented):
This is untested but I think something like this would be nice:
https://github.com/martonp/dcrdex/commit/9f4186faaf110240413650861491be86f9936e9b

Then the logic of the `pokesCache` could be unit tested.

peterzen commented:
> This is untested but I think something like this would be nice: [martonp@9f4186f](https://github.com/martonp/dcrdex/commit/9f4186faaf110240413650861491be86f9936e9b)
> 
Thanks, incorporated this refactor and added tests.  Simplified `pokes()` to sort the pokes on timestamp.

peterzen commented:
> When I make code suggestions, they're usually not well-tested. Regardless, you should test the functionality of everything you submit.

These are now fixed.  That was very lousy testing on my part, apologies.

JoeGruffins commented on client/db/bolt/db_test.go:
```diff
@@ -1619,3 +1619,39 @@ func TestOrderSide(t *testing.T) {
+	}
+
+	if err := boltdb.SavePokes(pokes); err != nil {
+		t.Fatalf("Save3Pokes error: %v", err)
```
```suggestion
		t.Fatalf("SavePokes error: %v", err)
```
